packages/block-editor/src/components/block-list/block-wrapper.js

```diff
@@ -265,6 +275,55 @@ export function useBlockWrapperProps( props = {}, { __unstableIsHtml } = {} ) {
 		};
 	}, [ isNavigationMode, isHovered, setHovered ] );
 
+	useEffect( () => {
+		if ( isFullSiteEditingActive ) {
+			// To accurately determine which blocks are hovered we must look at the elements under the cursor.
+			// Internal block inserters will fire events like mouseleave for blocks they are visually contained within,
+			// which makes an add/remove by clientId per individual block approach inaccurate.
+			function getHoveredBlocksFromCursor( event ) {
+				const hoveredElements = document.elementsFromPoint(
+					event.clientX,
+					event.clientY
+				);
+				const blockIds = [];
+				hoveredElements.forEach( ( element ) => {
+					if ( element.dataset.block ) {
+						blockIds.push( element.dataset.block );
+					}
+				} );
+				return blockIds;
+			}
+			function evaluateHoveredBlocks( event ) {
+				// Check the timeStamp of the last time this was set.
+				// This prevents needlessly rerunning the parse and dispatch for nested blocks
+				// who share boundaries that trigger the mouse events at or around the same time.
+				if (
+					event.timeStamp - ( getHoveredBlocksTimeStamp() || 0 ) >
+					10
+				) {
```

Comment:
So this is bugging me.  Of course different browsers can't handle events consistently 🤦‍♀️  .

So Chrome will make `event.timeStamp` the same for all events created at the same time.  Firefox handles the first event fully and then creates the second one, so the timeStamps end up being later based on how long the first took to run.  In practice with running the parse and dispatch, thats ~5ms.  Additionally, Firefox dev tools doesn't seem to have a way to CPU throttle, so testing this for slower CPUs isn't straight forward in that browser.  

So if we want this fix for not unnecessarily running the evaluation to work on firefox we need to do `timeStamp1 - teimpStamp2 > someDelta` instead of `timeStamp1 !== timeStamp2`.  But from what I saw previously (in testing on chrome) this parse can take between 40-70ms to run on a slow (x6 slowdown processor).  If we make this delta something like 70-100ms, we start experiencing funky behavior when moving the cursor out of blockA and into blockB sometimes not re-evaluating what is hovered (although still probably better than the inserter disrupting the hover state).

🤷‍♀️  😭  ❓ 😠  not sure what to think at this point.

Reply:
creating a new timeStamp just before the dispatch may help, I will evaluate this a bit more tomorrow.

Reply:
Yep, this seems to take care of things.